Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -738,36 +783,36 @@ private async void QueryResults()
 
             _lastQuery = query;
 
-            var plugins = PluginManager.ValidPluginsForQuery(query);
-
-            if (plugins.Count == 1)
-            {
-                PluginIconPath = plugins.Single().Metadata.IcoPath;
-                SearchIconVisibility = Visibility.Hidden;
-            }
-            else
-            {
-                PluginIconPath = null;
-                SearchIconVisibility = Visibility.Visible;
-            }
 
             _ = Task.Delay(200, currentCancellationToken).ContinueWith(_ =>
-            {
-                // start the progress bar if query takes more than 200 ms and this is the current running query and it didn't finish yet
-                if (!currentCancellationToken.IsCancellationRequested && _isQueryRunning)
                 {
-                    ProgressBarVisibility = Visibility.Visible;
-                }
-            }, currentCancellationToken, TaskContinuationOptions.NotOnCanceled, TaskScheduler.Default);
+                    // start the progress bar if query takes more than 200 ms and this is the current running query and it didn't finish yet
+                    if (!currentCancellationToken.IsCancellationRequested && _isQueryRunning)
+                    {
+                        ProgressBarVisibility = Visibility.Visible;
+                    }
+                },
+                currentCancellationToken,
+                TaskContinuationOptions.NotOnCanceled,
+                TaskScheduler.Default);
 
             try
             {
-                await Parallel.ForEachAsync(plugins, currentCancellationToken,
+                await Parallel.ForEachAsync(plugins,
+                    currentCancellationToken,
                     async (pair, token) =>
                     {
-                        await Task.Delay(pair.Metadata.SearchDelay ?? Settings.SearchDelay, token);
-                        if (pair.Metadata.Disabled)
+                        if(token.IsCancellationRequested)
+                            return;
+
+                        await Task.Delay(pair.Metadata.SearchDelay
+                                         ?? (string.IsNullOrEmpty(query.ActionKeyword)
+                                             ? Settings.GlobalSearchDelay
+                                             : 0));
```

Comment:
Is it possible for plugins to notify that should use user specified delay value?

Reply:
nvm

Reply:
And why use delay only for global searches?